frontend/src/pages/modelRegistry/screens/RegisteredModels/usePrefillDeployModalFromModelRegistry.ts

```diff
@@ -1,35 +1,44 @@
 import { AlertVariant } from '@patternfly/react-core';
 import React from 'react';
+import { Connection } from '~/concepts/connectionTypes/types';
 import { ProjectKind } from '~/k8sTypes';
-import useLabeledDataConnections from '~/pages/modelRegistry/screens/RegisteredModels/useLabeledDataConnections';
 import { RegisteredModelDeployInfo } from '~/pages/modelRegistry/screens/RegisteredModels/useRegisteredModelDeployInfo';
 import {
   CreatingInferenceServiceObject,
   InferenceServiceStorageType,
-  LabeledDataConnection,
+  LabeledConnection,
 } from '~/pages/modelServing/screens/types';
 import { AwsKeys, EMPTY_AWS_SECRET_DATA } from '~/pages/projects/dataConnections/const';
-import useDataConnections from '~/pages/projects/screens/detail/data-connections/useDataConnections';
-import { DataConnection, UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import useConnections from '~/pages/projects/screens/detail/connections/useConnections';
+import useLabeledConnections from './useLabeledConnections';
 
 const usePrefillDeployModalFromModelRegistry = (
-  projectContext: { currentProject: ProjectKind; dataConnections: DataConnection[] } | undefined,
+  projectContext: { currentProject: ProjectKind; connections: Connection[] } | undefined,
   createData: CreatingInferenceServiceObject,
   setCreateData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>,
   registeredModelDeployInfo?: RegisteredModelDeployInfo,
-): [LabeledDataConnection[], boolean, Error | undefined] => {
-  const [fetchedDataConnections, dataConnectionsLoaded, dataConnectionsLoadError] =
-    useDataConnections(projectContext ? undefined : createData.project);
-  const allDataConnections = projectContext?.dataConnections || fetchedDataConnections;
-  const { dataConnections, storageFields } = useLabeledDataConnections(
+): [LabeledConnection[], boolean, Error | undefined] => {
+  const [fetchedConnections, connectionsLoaded, connectionsLoadError] = useConnections(
+    projectContext ? undefined : createData.project,
+    true,
+  );
+  const allConnections = projectContext?.connections || fetchedConnections;
+  const { connections, storageFields } = useLabeledConnections(
     registeredModelDeployInfo?.modelArtifactUri,
-    allDataConnections,
+    allConnections,
   );
 
   React.useEffect(() => {
-    if (registeredModelDeployInfo) {
+    const alert = {
+      type: AlertVariant.info,
+      title: "We've auto-switched to create a new connection and pre-filled the details for you.",
+      message:
+        'Model location info is available in the registry but there are no matching connections in the project. So we automatically switched the option to create a new connection and prefilled the information.',
+    };
+    if (registeredModelDeployInfo?.modelArtifactUri) {
```

Comment:
This section of nested conditionals is quite hard to grok, I think it might be worth seeing if this can be simplified, anyone who needs to enhance or fix bugs in this area in the future could struggle with updating this. 

I'll take a deeper look and see if I can suggest anything.

Reply:
This could be done after merging if it's going to hold things up, btw.